code/__HELPERS/radiation.dm

```diff
@@ -0,0 +1,56 @@
+// A special GetAllContents that doesn't search past things with rad insulation
+// Components which return COMPONENT_BLOCK_RADIATION prevent further searching into that object's contents. The object itself will get returned still.
+// The ignore list makes those objects never return at all
+/proc/get_rad_contents(atom/location)
+	var/static/list/ignored_things = typecacheof(list(
+		/mob/dead,
+		/mob/camera,
+		/obj/effect,
+		/obj/docking_port,
+		/atom/movable/lighting_object,
+		/obj/item/projectile,
+		))
+	var/list/processing_list = list(location)
+	. = list()
+	while(processing_list.len)
+		var/atom/thing = processing_list[1]
+		processing_list -= thing
+		if(ignored_things[thing.type])
+			continue
+		. += thing
+		if((thing.flags_2 & RAD_PROTECT_CONTENTS_2) || (SEND_SIGNAL(thing, COMSIG_ATOM_RAD_PROBE) & COMPONENT_BLOCK_RADIATION))
+			continue
+		processing_list += thing.contents
+
+/proc/radiation_pulse(atom/source, intensity, range_modifier, log = FALSE, can_contaminate = TRUE)
+	if(!SSradiation.can_fire)
```

Comment:
```suggestion
	if(!SSradiation.flags & SS_NO_FIRE)
```

Subsystems are given the `SS_NO_FIRE` flag when offlined, not `can_fire = FALSE`.
```dm
BadBoy.flags |= SS_NO_FIRE
```

Reply:
this is meant for when the SS is intentionally halted, not so much when offlined:

```
	//set to 0 to prevent fire() calls, mostly for admin use or subsystems that may be resumed later
	//	use the SS_NO_FIRE flag instead for systems that never fire to keep it from even being added to the list
	var/can_fire = TRUE	
```